src/main/java/io/cdap/plugin/sap/SapODataConfig.java

```diff
@@ -0,0 +1,251 @@
+      }
+    }
+    if (!containsMacro(SapODataConstants.ODATA_SERVICE_URL) && Strings.isNullOrEmpty(url)) {
+      throw new InvalidConfigPropertyException("OData Service URL must be specified",
+                                               SapODataConstants.ODATA_SERVICE_URL);
+    }
+    if (!containsMacro(SapODataConstants.RESOURCE_PATH) && Strings.isNullOrEmpty(resourcePath)) {
+      throw new InvalidConfigPropertyException("Resource path must be specified", SapODataConstants.RESOURCE_PATH);
+    }
+
+    if (!Strings.isNullOrEmpty(schema) && !containsMacro(SapODataConstants.SCHEMA)) {
+      Schema parsedSchema = getParsedSchema();
+      validateSchema(parsedSchema);
+    }
+  }
+
+  private void validateSchema(Schema parsedSchema) {
+    List<Schema.Field> fields = parsedSchema.getFields();
+    if (null == fields || fields.isEmpty()) {
+      throw new InvalidConfigPropertyException("Schema should contain fields to map", SapODataConstants.SCHEMA);
+    }
+    for (Schema.Field field : fields) {
+      Schema nonNullableSchema = field.getSchema().isNullable() ?
+        field.getSchema().getNonNullable() : field.getSchema();
+      Schema.Type type = nonNullableSchema.getType();
+      Schema.LogicalType logicalType = nonNullableSchema.getLogicalType();
+      if (!SUPPORTED_SIMPLE_TYPES.contains(type) && !SUPPORTED_LOGICAL_TYPES.contains(logicalType)) {
+        String supportedTypeNames = Stream.concat(SUPPORTED_SIMPLE_TYPES.stream(), SUPPORTED_LOGICAL_TYPES.stream())
+          .map(Enum::name)
+          .map(String::toLowerCase)
+          .collect(Collectors.joining(", "));
+        String actualTypeName = logicalType != null ? logicalType.name().toLowerCase() : type.name().toLowerCase();
+        String errorMessage = String.format("Field '%s' is of unsupported type '%s'. Supported types are: %s.",
+                                            field.getName(), actualTypeName, supportedTypeNames);
+        throw new IllegalArgumentException(errorMessage);
+      }
+    }
+  }
+
+  /**
+   * Validate that the provided schema is compatible with the inferred schema. The provided schema is compatible if
+   * every field is compatible with the corresponding field in the inferred schema. A field is compatible if it is of
+   * the same type or is a nullable version of that type. It is assumed that both schemas are record schemas.
+   *
+   * @param inferredSchema the inferred schema
+   * @param providedSchema the provided schema to check compatibility
+   * @throws IllegalArgumentException if the schemas are not type compatible
+   */
+  public static void validateFieldsMatch(Schema inferredSchema, Schema providedSchema) {
+    for (Schema.Field field : providedSchema.getFields()) {
+      Schema.Field inferredField = inferredSchema.getField(field.getName());
+      if (inferredField == null) {
+        throw new IllegalArgumentException(String.format("Field '%s' does not exist.", field.getName()));
```

Comment:
this doesn't need to be an error. We should allow the user to configure a schema that is a subset of the actual schema,

Reply:
Here we iterate over provided schema fields, so the exception will be thrown if the actual schema does not contain the specified field(changed to use `FailureCollector`).